src/boot/efi/boot.c

```diff
@@ -1651,6 +1651,51 @@ static EFI_STATUS image_start(EFI_HANDLE parent_image, const Config *config, con
                 loaded_image->LoadOptionsSize = (StrLen(loaded_image->LoadOptions)+1) * sizeof(CHAR16);
 
 #ifdef SD_BOOT_LOG_TPM
+                UINTN i, j;
+                UINTN length;
+                CHAR16* begin, *end;
+
+                EFI_FILE* root_dir = LibOpenRoot(entry->device);
+                if (!root_dir) {
+                        Print(L"Unable to open root directory: %r ", err);
+                        uefi_call_wrapper(BS->Stall, 1, 3 * 1000 * 1000);
+                        return EFI_LOAD_ERROR;
+                }
+
+                length = StrLen(options);
+                for (i = 0; i < length; i++) {
+                        if (StrnCmp(L"initrd=", &options[i], 7) == 0) {
+                                begin = &options[i + 7];
+                                for (j = i + 7; j < length && options[j] != L' '; j++);
+                                end = &options[j];
+                                UINTN initrd_path_len = (end - begin) * sizeof(CHAR16);
```

Comment:
no variable declarations in the middle of code blocks. (here and evreywhere else, see above/CODING_STYLE)